receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -83,13 +97,48 @@ func (ghs *githubScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 		ghs.logger.Sugar().Debugf("using search query where query is: %v", ghs.cfg.SearchQuery)
 	}
 
-	repos, count, err := ghs.getRepos(ctx, genClient, sq)
-	if err != nil {
-		ghs.logger.Sugar().Errorf("error getting repo data: %v", zap.Error(err))
-		return ghs.mb.Emit(), err
+	var repos []unifiedRepo
+	var repoCount int
+
+	// If both team and search query are set, team takes precedence.  Might want to make it
+	// impossible to set both in the config
+	if ghs.cfg.GitHubTeam != "" {
+		teamRepos, count, err := ghs.getTeamRepos(ctx, genClient)
+		if err != nil {
+			ghs.logger.Sugar().Errorf("error getting repo data: %v", zap.Error(err))
+			return ghs.mb.Emit(), err
+		}
+		repoCount = count
+		for _, repo := range teamRepos {
+			unifiedRepo := unifiedRepo{
+				Id:   repo.Id,
+				Name: repo.Name,
+				DefaultBranchRef: defaultBranch{
+					Name: repo.DefaultBranchRef.Name,
+				},
+			}
+			repos = append(repos, unifiedRepo)
+		}
+	} else {
+		searchRepos, count, err := ghs.getSearchRepos(ctx, genClient, sq)
+		if err != nil {
+			ghs.logger.Sugar().Errorf("error getting repo data: %v", zap.Error(err))
+			return ghs.mb.Emit(), err
+		}
+		repoCount = count
+		for _, repo := range searchRepos {
+			unifiedRepo := unifiedRepo{
+				Id:   repo.Id,
+				Name: repo.Name,
+				DefaultBranchRef: defaultBranch{
+					Name: repo.DefaultBranchRef.Name,
+				},
+			}
+			repos = append(repos, unifiedRepo)
+		}
 	}
 
-	ghs.mb.RecordGitRepositoryCountDataPoint(now, int64(count))
+	ghs.mb.RecordGitRepositoryCountDataPoint(now, int64(repoCount))
```

Comment:
See my above under the hood comment. I'd like to see this maintain it's `count` not repo count value.